python-backend/app/api/mines/mine/resources/mine.py

```diff
@@ -436,17 +451,19 @@ def _mine_status_processor(mine_status, mine):
 
     existing_status = mine.mine_status[0] if mine.mine_status else None
     if existing_status:
-        if existing_status.mine_status_xref_guid == mine_status_xref.mine_status_xref_guid:
+        if existing_status.mine_status_xref_guid == mine_status_xref.mine_status_xref_guid \
+                and status_date == existing_status.status_date:
             return existing_status
 
         existing_status.expiry_date = datetime.today()
         existing_status.active_ind = False
         existing_status.save()
-
-    new_status = MineStatus(mine_status_xref_guid=mine_status_xref.mine_status_xref_guid)
+    if not status_date:
+        new_status = MineStatus(mine_status_xref_guid=mine_status_xref.mine_status_xref_guid)
+    else:
+        new_status = MineStatus(mine_status_xref_guid=mine_status_xref.mine_status_xref_guid, status_date=status_date)
```

Comment:
not sure why this conditional is here. status_date is nullable.

Reply:
good catch, removed